src/Authenticator/TokenAuthenticator.php

```diff
@@ -30,17 +30,65 @@ class TokenAuthenticator extends AbstractAuthenticator
      * @param \Psr\Http\Message\ServerRequestInterface $request The request that contains login information.
      * @return string|null
      */
-    protected function _getToken(ServerRequestInterface $request)
+    protected function getToken(ServerRequestInterface $request)
     {
-        $header = $this->config('header');
-        if (!empty($header)) {
-            return $request->getHeaderLine($header);
+        $token = $this->getTokenFromHeader($request, $this->getConfig('header'));
+        if (empty($token)) {
+            $token = $this->getTokenFromQuery($request, $this->getConfig('queryParam'));
+        }
+
+        $prefix = $this->getConfig('tokenPrefix');
+        if (is_string($token) && !empty($prefix)) {
+            return $this->stripTokenPrefix($token, $prefix);
         }
 
-        $param = $this->config('queryParam');
+        return $token;
+    }
+
+    /**
+     * Strips a prefix from a token
+     *
+     * @param string $token Token string
+     * @param string $prefix Prefix to strip
+     * @return string
+     */
+    protected function stripTokenPrefix($token, $prefix)
+    {
+        return str_ireplace($prefix . ' ', '', $token);
+    }
+
+    /**
+     * Gets the token from the request headers
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request The request that contains login information.
+     * @param string $headerLine Header name
+     * @return string|null
+     */
+    protected function getTokenFromHeader(ServerRequestInterface $request, $headerLine)
+    {
+        if (!empty($headerLine)) {
+            $header = $request->getHeaderLine($headerLine);
+            if (!empty($header)) {
+                return $request->getHeaderLine($headerLine);
```

Comment:
Why are you calling `getHeaderLine` again? Couldn't you do `return $header` here?